### PR TITLE
Fix duplicate ZEIR IDs

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes bug causing duplicate ZEIR IDs and removes duplicates by generating assigning un-assigned IDs https://github.com/opensrp/support/issues/30

### CODE REVIEW

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/interactor/LoginInteractor.java [hunk 1]
```diff
@@ -69,5 +69,11 @@ protected void scheduleJobsImmediately() {
         PullUniqueIdsServiceJob.scheduleJobImmediately(PullUniqueIdsServiceJob.TAG); //need these asap!
         ZScoreRefreshIntentServiceJob.scheduleJobImmediately(ZScoreRefreshIntentServiceJob.TAG);
         ImageUploadServiceJob.scheduleJobImmediately(ImageUploadServiceJob.TAG);
+
+        // This job will not be duplicated but is added here since scheduleJobsPeriodically is only called
+        // after a remote login and therefore might be run too late. scheduleJobsImmediately is called
+        // after both remote login and local login
```
Comment (Rkareko) [inline opensrp-giz-malawi/src/main/java/org/smartregister/giz/interactor/LoginInteractor.java:75]: This looks like a single comment. Could we use a block comment.

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ChildDetailTabbedActivity.java [hunk 2]
```diff
@@ -248,5 +248,10 @@ protected String getReportDeceasedMetadata() {
         }
         return "";
     }
+
+    @Override
+    public void updateViews() {
+        // Do nothing
+    }
```
Comment (Rkareko) [inline opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ChildDetailTabbedActivity.java:255]: Why is this being overriden?

File: opensrp-giz-malawi/src/test/java/org/smartregister/giz/repository/GizMalawiRepositoryTest.java [hunk 3]
```diff
@@ -49,4 +53,37 @@ public void onCreateShouldCreate62tables() {
         gizMalawiRepository.onCreate(database);
         Mockito.verify(database, Mockito.times(63)).execSQL(Mockito.contains("CREATE TABLE"));
     }
+
+    @Test
+    public void onUpgradeFrom19To20ShouldCreateVaccineOverdueCountRepositoryAndMigrateVaccines() {
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{"id", "name"});
+        //matrixCursor.moveToNext()
```
Comment (Rkareko) [inline opensrp-giz-malawi/src/test/java/org/smartregister/giz/repository/GizMalawiRepositoryTest.java:60]: Delete this

File: opensrp-giz-malawi/src/test/java/org/smartregister/giz/repository/GizMalawiRepositoryTest.java [hunk 4]
```diff
@@ -49,4 +53,37 @@ public void onCreateShouldCreate62tables() {
...
+
+        Mockito.verify(database).execSQL(VaccineOverdueCountRepository.CREATE_TABLE_SQL);
+        Mockito.verify(database).execSQL(VaccineOverdueCountRepositoryHelper.MIGRATE_VACCINES_QUERY);
+    }
+
+    @Test
+    public void onUpgradeFrom19To20ShouldAddOutreachColumn() {
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{"id", "name"});
+        //matrixCursor.moveToNext()
+        SQLiteDatabase database = Mockito.mock(SQLiteDatabase.class);
```
Comment (Rkareko) [inline opensrp-giz-malawi/src/test/java/org/smartregister/giz/repository/GizMalawiRepositoryTest.java:74]: Delete this if not required

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ChildDetailTabbedActivity.java (hunk 2 above)
Comment (ekigamba) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ChildDetailTabbedActivity.java:255]: It's abstract and needs to be implemented

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/interactor/LoginInteractor.java (hunk 1 above)
Comment (ekigamba) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/interactor/LoginInteractor.java:75]: Definitely

File: opensrp-giz-malawi/src/test/java/org/smartregister/giz/repository/GizMalawiRepositoryTest.java (hunk 4 above)
Comment (ekigamba) [reply opensrp-giz-malawi/src/test/java/org/smartregister/giz/repository/GizMalawiRepositoryTest.java:74]: Nice catch

**Review by Rkareko — APPROVED** [review]

**Pull request merged**